apps/aecontract/src/aect_channel_contract.erl

```diff
@@ -21,28 +21,21 @@ new(Owner, Round, VmVersion, Code, Deposit, Trees0) ->
     {ContractId, Contract, Trees1}.
 
 -spec run_new(aect_contracts:pubkey(), aect_call:call(), binary(),
-              non_neg_integer(), aec_trees:trees())
-    -> aec_trees:trees().
-run_new(ContractPubKey, Call, CallData, Round, Trees0) ->
+              aec_trees:trees(), aec_trees:trees(),
+              aetx_env:env()) -> aec_trees:trees().
+run_new(ContractPubKey, Call, CallData, Trees0, OnChainTrees,
+        OnChainEnv) ->
     ContractsTree  = aec_trees:contracts(Trees0),
     Contract = aect_state_tree:get_contract(ContractPubKey, ContractsTree),
     OwnerPubKey = aect_contracts:owner_pubkey(Contract),
     Code = aect_contracts:code(Contract),
     CallStack = [], %% TODO: should we have a call stack for create_tx also
                     %% when creating a contract in a contract.
     VmVersion = aect_contracts:vm_version(Contract),
-    CallDef = #{ caller      => OwnerPubKey
-               , contract    => ContractPubKey
-               , gas         => 10000000
-               , gas_price   => 1
-               , call_data   => CallData
-               , amount      => 0
-               , call_stack  => CallStack
-               , code        => Code
-               , call        => Call
-               , trees       => Trees0
-               , tx_env     => tx_env(Round)
-               },
+    CallDef = make_call_def(OwnerPubKey, ContractPubKey,
+                            _Gas = 1000000, _GasPrice = 1,
+                            _Amount = 0, %TODO: make this configurable
```

Comment:
I guess the TODO includes Gas and GasPrice.

Reply:
I understand this is never used on-chain but only offchain, right? So gas price may even be 0?

Reply:
(While `run` is used on-chain for force progress tx.)

Reply:
Yes, TODO includes amount, gas limit and gas price althogether. All three are part of the in-chain protocol (`aesc_offchain_udpate`'s call `udpate`) but they are not exposed to WebSocket API. Adding the API after mainnet launch will not impact protocol.
Current values are dummy ones and yes, since gas is not consumed off-chain, it could as well be 0. Those values are important for on-chain execution (where gas is consumed and `gas_price` is important)